opus/application/static_media/js/browse.js

```diff
@@ -548,34 +521,102 @@ var o_browse = {
                         o_browse.removeEditMetadataDetails();
                         o_browse.loadPageIfNeeded("prev", detailOpusId);
                         break;
+                    case 38:  // up
+                        offset = (o_browse.isGalleryView() ? viewNamespace.galleryBoundingRect.x : 1);
+                        if (obsNum > offset) {
+                            obsNum -= offset;
+                            detailOpusId = (o_browse.isGalleryView() ? $(tab).find(`.op-thumbnail-container[data-obs='${obsNum}']`).data("id") : $(tab).find(`tr[data-obs='${obsNum}']`).data("id"));
+                            o_browse.removeEditMetadataDetails();
+                            o_browse.loadPageIfNeeded("prev", detailOpusId);
+                        }
+                        break;
+                    case 40:  // down
+                        offset = (o_browse.isGalleryView() ? viewNamespace.galleryBoundingRect.x : 1);
+                        if (obsNum + offset <= viewNamespace.totalObsCount) {
+                            obsNum += offset;
+                            // if this new observation is in the last row of screen, don't allow the user to arrow there
+                            if (!o_browse.isLastRowOfObservations(tab, viewNamespace, obsNum)) {
+                                detailOpusId = (o_browse.isGalleryView() ? $(tab).find(`.op-thumbnail-container[data-obs='${obsNum}']`).data("id") : $(tab).find(`tr[data-obs='${obsNum}']`).data("id"));
+                                o_browse.removeEditMetadataDetails();
+                                o_browse.loadPageIfNeeded("next", detailOpusId);
+                            } else {
+                                return false;
+                            }
+                        }
+                        break;
                 }
                 if (detailOpusId && !$("#galleryViewContents").hasClass("op-disabled")) {
-                    o_browse.updateGalleryView(detailOpusId);
+                    o_browse.updateGalleryView(detailOpusId, obsNum);
+                    return false;
                 }
             }
             // don't return false here or it will snatch all the user input!
         });
     }, // end browse behaviors
 
+    isLastRowOfObservations: function(tab, viewNamespace, obsNum) {
+        let firstObsOnScreen = $(`${tab} .op-observation-slider`).slider("option", "value");
+        let obsPerRow = viewNamespace.galleryBoundingRect.x;
+        let totalObsOnScreen = obsPerRow * viewNamespace.galleryBoundingRect.yFloor;
+        let beginningOfLastRow = firstObsOnScreen + (totalObsOnScreen - obsPerRow);
+        let lastObs = $(`${tab} .op-thumbnail-container`).last().data("obs");
+        return (lastObs - firstObsOnScreen < o_browse.getLimit() && obsNum >= beginningOfLastRow);
+    },
+
+    onGalleryOrRowClick: function(obj, e) {
+        // make sure selected modal thumb is unhighlighted, as clicking on this closes the modal
```

Comment:
I really hate the use of the word "modal" everywhere. There are dozens of "modals" in OPUS. We should always specify which one we're talking about. Do you mean the "slide show modal"? (I know this was an existing comment that got moved, but still we should fix it)